sync/class.jetpack-sync-queue.php

```diff
@@ -50,13 +50,20 @@ function add( $item ) {
 			$rows_added = $wpdb->query( $wpdb->prepare(
 				"INSERT INTO $wpdb->options (option_name, option_value, autoload) VALUES (%s, %s,%s)",
 				$this->get_next_data_row_option_name(),
-				serialize( $item ),
+				$this->serialize( $item ),
 				'no'
 			) );
 			$added      = ( 0 !== $rows_added );
 		}
 	}
 
+	function serialize( $item ) {
```

Comment:
The biggest question I have is how can we make this more generic than testing for a SimpleXMLElement object? Which I suppose also brings up the question of, "Do we need to make it more generic"?